src/System.Net.Http/tests/FunctionalTests/LoopbackGetRequestHttpProxy.cs

```diff
@@ -0,0 +1,128 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Net.Http.Functional.Tests
+{
+    /// <summary>
+    /// Provides a test-only, overly-simplified HTTP proxy for a single GET request.  It is not meant to be robust,
+    /// but simply for verification that a GET request is in fact getting routed through a designated proxy
+    /// and providing credentials if expected.
+    /// </summary>
+    internal sealed class LoopbackGetRequestHttpProxy
```

Comment:
You should add a comment that this only supports non-SSL GET requests.  If the destination server was SSL, then usually the "CONNECT" verb is passed to the proxy.